library/sha256.c

```diff
@@ -287,7 +287,8 @@ int mbedtls_internal_sha256_process( mbedtls_sha256_context *ctx,
     {
         return( 0 );
     }
-
+    /* Free the ctx upon suspected FI */
+    mbedtls_sha256_free( ctx );
```

Comment:
I think that kind of operation should be mentioned in the description of the caller function.

Reply:
Normally I would agree, but in this case it will only happen when a fault injection is detected. We would have to update a whole lot of functions that do not describe their behavior in such situation at all.